### PR TITLE
minor mods: use the bool operator on ref_t and co.

### PR DESCRIPTION
**Pull request by 0xeb**

### CODE REVIEW

**Comment by 0xeb** [comment]
@aundro a second pair of eyes on this would be appreciated.

File: idapython.cpp [hunk 1]
```diff
@@ -1129,8 +1129,7 @@ ref_t idapython_plugin_t::get_sys_displayhook()
   ref_t h;
   if ( config.repl_use_sys_displayhook )
   {
-    ref_t py_sys(PyW_TryImportModule("sys"));
-    if ( py_sys != nullptr )
+    if ( ref_t py_sys = ref_t(PyW_TryImportModule("sys")) )
```
Comment (aundro) [inline idapython.cpp:1132]: Unless I'm mistaken, this implies C++-17 (conditional with initializer). 
Should we really impose that? (it's minor syntactic sugar)

Also, this doesn't seem to be harmonized across the entire patch (see the previous chunk)

File: pywraps.hpp [hunk 2]
```diff
@@ -1200,9 +1200,9 @@ struct hooks_base_t
       ref_t py_id;
       if ( PyObject_HasAttrString(self, "id") )
         py_id = newref_t(PyObject_GetAttrString(self, "id"));
-      if ( py_id == nullptr || !PyUnicode_Check(py_id.o) )
+      if ( !py_id || !PyUnicode_Check(py_id.o) )
         py_id = newref_t(PyObject_Repr(self));
-      if ( py_id != nullptr && PyUnicode_Check(py_id.o) )
+      if ( !py_id && PyUnicode_Check(py_id.o) )
         PyUnicode_as_qstring(&identifier, py_id.o);
```
Comment (aundro) [inline pywraps.hpp:1206]: This one seems wrong

File: pywraps/py_kernwin.hpp [hunk 3]
```diff
@@ -1188,7 +1190,7 @@ uint32 py_call_nav_colorizer(
     return 0;
   borref_t py_fun(PyDict_GetItemString(dict, "fun"));
   borref_t py_ud(PyDict_GetItemString(dict, "ud"));
-  if ( py_fun == nullptr
+  if ( py_fun
     || !PyCapsule_IsValid(py_fun.o, VALID_CAPSULE_NAME)
```
Comment (aundro) [inline pywraps/py_kernwin.hpp:1194]: This one looks wrong

**Review by aundro — COMMENTED** [review]
Besides the 2 bugs and the open question about C++-17, it's looking good, thank you!

**Comment by 0xeb** [comment]
@aundro , I fixed the two issues.

Yes, why not have C++17 requirements. All up to date Mac systems have this support by default. VS2019 supports it too, etc.


I used the conditional initializers where it made sense and where the line length was still reasonable.

**Pull request merged**